src/EditorFeatures/VisualBasicTest/CodeActions/InvertIf/InvertIfTests.vb

```diff
@@ -0,0 +1,1353 @@
+"Module Program
+    Sub Main()
+        [||]If a Then
+            a()
+#ExternalSource File.vb 1 
+        Else
+#End ExternalSource
+            b()
+        End If
+    End Sub
+End Module")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)>
+        Public Async Function TestMissingOnNonEmptyTextSpan() As Task
+            Await TestMissingInRegularAndScriptAsync(
+"Module Program
+    Sub Main()
+        [|If a Th|]en a() Else b()
+    End Sub
+End Module")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)>
+        Public Async Function TestMultipleStatementsSingleLineIfStatement() As Task
+            Await TestInRegularAndScriptAsync(
+"Module Program
+    Sub Main()
+        If[||] a Then a() : b() Else c() : d()
+    End Sub
+End Module",
+"Module Program
+    Sub Main()
+        If Not a Then c() : d() Else a() : b()
+    End Sub
+End Module")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)>
+        Public Async Function TestMultipleStatementsMultiLineIfBlock() As Task
+            Await TestInRegularAndScriptAsync(
+"Module Program
+    Sub Main()
+        [||]If a Then
+            goo()
+            bar()
+        Else
+            you()
+            too()
+        End If
+    End Sub
+End Module",
+"Module Program
+    Sub Main()
+        If Not a Then
+            you()
+            too()
+        Else
+            goo()
+            bar()
+        End If
+    End Sub
+End Module")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)>
+        Public Async Function TestTriviaAfterSingleLineIfStatement() As Task
+            Await TestInRegularAndScriptAsync(
+"Module Program
+    Sub Main()
+        [||]If a Then a() Else b() ' I will stay put 
+    End Sub
+End Module",
+"Module Program
+    Sub Main()
+        If Not a Then b() Else a() ' I will stay put 
+    End Sub
+End Module")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)>
+        Public Async Function TestTriviaAfterMultiLineIfBlock() As Task
+            Await TestInRegularAndScriptAsync(
+"Module Program
+    Sub Main()
+        [||]If a Then
+            a()
+        Else
+            b()
+        End If ' I will stay put 
+    End Sub
+End Module",
+"Module Program
+    Sub Main()
+        If Not a Then
+            b()
+        Else
+            a()
+        End If ' I will stay put 
+    End Sub
+End Module")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)>
+        Public Async Function TestParenthesizeForLogicalExpressionPrecedence() As Task
+            Await TestInRegularAndScriptAsync(
+"Sub Main()
+    I[||]f a AndAlso b Or c Then a() Else b()
+End Sub
+End Module",
+"Sub Main()
+    If (Not a OrElse Not b) And Not c Then b() Else a()
```

Comment:
yeesh.  i feel like this is actually worse.  but i'm not sure hte best thing to do about it.  But ```Not (a AndAlso b Or c``` is just much nicer here.  Have we considered comparing the size of the expression before/after propogating the 'not' downwards to make sure ew pick the smaller of the two?